addons/portal/models/res_partner.py

```diff
@@ -18,3 +17,10 @@ def can_edit_vat(self):
         edit it (as in backend)."""
         self.ensure_one()
         return not self.parent_id
+
+    @api.model
+    def _get_portal_partner_from_context(self):
```

Comment:
This code also does not really belong to partner model... it is a tool you can define in mail controllers as a classmethod and use it directly. Idea is to avoid putting all code in models, to avoid bloating business code with tooling used only in controllers (as all other code globally belong to controllers).

Reply:
This one will more than likely be used in various controllers too, which leads to a lot of dependency/inheritance annoyances when using it in practice.